pkg/pool-linear/contracts/aave/AaveLinearPoolFactory.sol

```diff
@@ -108,21 +139,20 @@ contract AaveLinearPoolFactory is
         address expectedRebalancerAddress = Create2.computeAddress(rebalancerSalt, keccak256(rebalancerCreationCode));
 
         (uint256 pauseWindowDuration, uint256 bufferPeriodDuration) = getPauseConfiguration();
-
-        AaveLinearPool.ConstructorArgs memory args = AaveLinearPool.ConstructorArgs({
-            vault: getVault(),
-            name: name,
-            symbol: symbol,
-            mainToken: mainToken,
-            wrappedToken: wrappedToken,
-            assetManager: expectedRebalancerAddress,
-            upperTarget: upperTarget,
-            swapFeePercentage: swapFeePercentage,
-            pauseWindowDuration: pauseWindowDuration,
-            bufferPeriodDuration: bufferPeriodDuration,
-            owner: owner,
-            version: getPoolVersion()
-        });
+        AaveLinearPool.ConstructorArgs memory args;
+
+        args.vault = getVault();
+        args.name = name;
+        args.symbol = symbol;
+        args.mainToken = mainToken;
+        args.wrappedToken = wrappedToken;
+        args.assetManager = expectedRebalancerAddress;
+        args.upperTarget = upperTarget;
+        args.swapFeePercentage = swapFeePercentage;
+        args.pauseWindowDuration = pauseWindowDuration;
+        args.bufferPeriodDuration = bufferPeriodDuration;
+        args.owner = owner;
+        args.version = getPoolVersion();
```

Comment:
One thing I like about initializing the struct inline is that it forces you to provide all the required fields. It is a bit more verbose / redundant, but a bit less error prone.

Is there a particular reason for this change?

Reply:
Yes - got a stack-too-deep as soon as I added the new parameter. This solved that (and is also more memory efficient)

Reply:
ugh :sad: we are on the edge with the stack pressure. Then I guess it's fine!